sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -57,7 +59,7 @@ public SentryTransaction(
     this.transaction = Objects.requireNonNull(name, "name is required");
     this.startTimestamp = DateUtils.getCurrentDateTime();
     this.hub = Objects.requireNonNull(hub, "hub is required");
-    this.getContexts().setTrace(contexts);
+    this.context = contexts;
```

Comment:
do the assignment with `Objects.requireNonNull(contexts, "contexts are required");`